providers/elasticsearch/src/airflow/providers/elasticsearch/hooks/elasticsearch.py

```diff
@@ -268,3 +282,448 @@ def search(self, query: dict[Any, Any], index: str = "_all") -> dict:
         es_client = self.get_conn
         result = es_client.search(index=index, body=query)
         return result["hits"]
+
+
+class ElasticsearchHook(BaseHook):
+    """
+    Hook for interacting with Elasticsearch.
+
+    This hook provides a convenient interface to Elasticsearch operations
+    including search, bulk operations, and data management.
+    """
+
+    conn_name_attr = "elasticsearch_conn_id"
+    default_conn_name = "elasticsearch_default"
+    conn_type = "elasticsearch"
+    hook_name = "Elasticsearch"
+
+    def __init__(
+        self, elasticsearch_conn_id: str = "elasticsearch_default", log_query: bool = False, **kwargs: Any
+    ) -> None:
+        """
+        Initialize the Elasticsearch Hook.
+
+        :param elasticsearch_conn_id: The Airflow connection ID for Elasticsearch.
+            Default: *elasticsearch_default*.
+        :param log_query: If *True*, queries will be logged for debugging purposes.
+            Default: *False*.
+        :param kwargs: Additional arguments passed to the parent BaseHook class.
+        """
+        import os
+
+        super().__init__()
+        self.conn_id = elasticsearch_conn_id
+        self.log_query = log_query
+
+        # Environment variables as fallback configuration
+        self.vars: dict[str, str | bool | None] = {
+            "host": os.getenv("ELASTICSEARCH_HOST"),
+            "port": os.getenv("ELASTICSEARCH_PORT"),
+            "username": os.getenv("ELASTICSEARCH_USERNAME"),
+            "password": os.getenv("ELASTICSEARCH_PASSWORD"),
+            "use_ssl": os.getenv("ELASTICSEARCH_USE_SSL", "false").lower() == "true",
+            "verify_certs": os.getenv("ELASTICSEARCH_VERIFY_CERTS", "true").lower() == "true",
+            "timeout": os.getenv("ELASTICSEARCH_TIMEOUT"),
+            "max_retries": os.getenv("ELASTICSEARCH_MAX_RETRIES"),
+        }
+
+    @cached_property
+    def conn(self) -> "AirflowConnection":
+        """Get the Airflow connection object for Elasticsearch."""
+        return self.get_connection(self.conn_id)
+
+    @cached_property
+    def client(self) -> Elasticsearch:
+        """
+        Create and return an Elasticsearch client with environment variable fallback.
+
+        Configuration priority:
+        1. Airflow connection (highest priority)
+        2. Environment variables (fallback)
+        3. Default values (last resort)
+
+        :return: Configured Elasticsearch client instance.
+        :raises AirflowConfigException: If client creation fails due to configuration issues.
+        """
+        # Get Airflow connection
+        conn = self.get_connection(self.conn_id)
+
+        # Configuration with fallback priority: connection -> env vars -> defaults
+        host = conn.host or self.vars["host"] or "localhost"
+        port = conn.port or (int(self.vars["port"]) if self.vars["port"] else None) or 9200
+        schema = conn.schema or ("https" if self.vars["use_ssl"] else "http")
```

Comment:
For consistency with `host` and `port`, the `schema` fallback could follow the same 
`conn → vars → default` structure, e.g.:

```suggestion
        schema = conn.schema or ("https" if self.vars.get("use_ssl") else None) or "http"
```